fraim/actions/github.py

```diff
@@ -0,0 +1,153 @@
+        RuntimeError: If GitHub token is not set or API calls fail
+    """
+    logger.info(f"Notifying GitHub user_or_group {user_or_group} for PR: {pr_url}")
+
+    # Parse PR URL to get owner, repo, and PR number
+    logger.debug(f"Parsing PR URL: {pr_url}")
+    try:
+        path_parts = urlparse(pr_url).path.strip("/").split("/")
+        if len(path_parts) < 4 or path_parts[-2] != "pull":
+            raise ValueError
+        owner, repo, _, pr_number = path_parts
+        logger.debug(f"Parsed PR details - Owner: {owner}, Repo: {repo}, PR: {pr_number}")
+    except (ValueError, IndexError):
+        logger.error(f"Failed to parse PR URL: {pr_url}")
+        raise ValueError(f"Invalid PR URL format: {pr_url}")
+
+    # Get GitHub token from multiple sources
+    logger.debug("Getting GitHub authentication token")
+    github_token = _get_github_token()
+    if not github_token:
+        logger.error("GitHub authentication failed - no token found")
+        raise RuntimeError(
+            "GitHub authentication required. Please ensure one of the following:\n"
+            "1. Set GITHUB_TOKEN environment variable\n"
+            "2. Configure git credentials for github.com\n"
+            "3. Login with GitHub CLI (`gh auth login`)"
+        )
+
+    # Initialize GitHub client
+    logger.debug("Initializing GitHub client")
+    gh = Github(github_token)
+
+    try:
+        # Get repository and PR
+        logger.debug(f"Getting repository: {owner}/{repo}")
+        repository = gh.get_repo(f"{owner}/{repo}")
+        logger.debug(f"Getting pull request #{pr_number}")
+        pull_request = repository.get_pull(int(pr_number))
+    except Exception as e:
+        logger.error(f"Failed to get repository or pull request: {str(e)}")
+        raise RuntimeError(f"Failed to get repository or pull request: {str(e)}")
+
+    try:
+        # Add comment with user_or_group mention
+        comment_text = f"{user_or_group}\n\n{description}"
+        logger.debug(f"Adding comment to PR with user_or_group mention: {user_or_group}")
+        pull_request.create_issue_comment(comment_text)
+        logger.info(f"Successfully added comment to PR #{pr_number}")
+    except Exception as e:
+        logger.error(f"Failed to add comment to PR: {str(e)}")
+        raise RuntimeError(f"Failed to add comment to PR: {str(e)}")
+
+    try:
+        # Request review from the group
+        # Strip @ from user_or_group name for the API
+        team_slug = user_or_group.strip("@").split("/")[-1]
+        logger.debug(f"Requesting review from user or group: {team_slug}")
+        pull_request.create_review_request(team_reviewers=[team_slug])
+        logger.info(f"Successfully requested review from user or group {team_slug} for PR #{pr_number}")
+    except Exception as e:
+        logger.error(f"Failed to request review from user or group: {str(e)}")
+        raise RuntimeError(f"Failed to request review from user or group: {str(e)}")
```

Comment:
```suggestion
        raise RuntimeError(f"Failed to request review from user or group: {str(e)}") from e
```
so that we don't lose the causing exceptino.